src/components/sbb-link/sbb-link.tsx

```diff
@@ -98,53 +180,34 @@ export class SbbLink {
       iconFlipClass = ' sbb-link--icon-flip';
     }
 
-    const variantClass = ` sbb-link--${this.variant}`;
-
-    /**
-     * Add additional attributes
-     * ----------------------------------------------------------------
-     */
-    let additionalLinkAttributes = {};
-    let ariaLabel = this.text;
+    const inlineClass = this.variant === 'inline' ? ' sbb-link--inline' : '';
+    const negativeClass = this.negative ? ' sbb-link--negative' : '';
 
-    if (openInNewWindow) {
-      additionalLinkAttributes = {
-        rel: 'external noopener nofollow',
-        target: '_blank',
-      };
-      ariaLabel += `. ${i18nTargetOpensInNewWindow[currentLanguage]}`;
-    }
+    return `sbb-link${textSizeClass}${iconPositionClass}${iconFlipClass}${inlineClass}${negativeClass}`;
+  }
 
-    if (this.idValue) {
-      additionalLinkAttributes = {
-        ...additionalLinkAttributes,
-        id: this.idValue,
-      };
+  /**
+   * Render element
+   */
+  public render(): JSX.Element {
+    let TAG_NAME;
+    if (this.isStatic) {
+      TAG_NAME = 'span';
+    } else if (this.href) {
+      TAG_NAME = 'a';
+    } else {
+      TAG_NAME = 'button';
     }
 
     return (
-      <a
-        aria-label={ariaLabel}
-        class={`sbb-link
-          ${textSizeClass}
-          ${iconPositionClass}
-          ${iconFlipClass}
-          ${variantClass}`}
-        download={this.download}
-        dir={currentWritingMode}
-        href={this.hrefValue}
-        {...additionalLinkAttributes}
-      >
-        {this.icon && !this._inlineVariant ? (
-          <span class="sbb-link__icon">
-            <slot name="icon" />
-          </span>
+      <TAG_NAME {...this._getAttributeList}>
+        {this.icon && this.variant !== 'inline' ? (
+          <sbb-icon name={this.icon}></sbb-icon>
```

Comment:
Also you can remove the check for "this.icon".
```suggestion
          <slot name='icon'>
            <sbb-icon name={this.icon}></sbb-icon>
          </slot>
```